src/VisualStudio/Core/Def/Implementation/Snippets/AbstractSnippetInfoService.cs

```diff
@@ -62,7 +62,9 @@ internal abstract class AbstractSnippetInfoService : ForegroundThreadAffinitized
         private async Task InitializeAndPopulateSnippetsCacheAsync(Shell.IAsyncServiceProvider asyncServiceProvider)
         {
             await _threadingContext.JoinableTaskFactory.SwitchToMainThreadAsync();
-            var textManager = (IVsTextManager2)await asyncServiceProvider.GetServiceAsync(typeof(SVsTextManager)).ConfigureAwait(true);
+            var textManager = (IVsTextManager2?)await asyncServiceProvider.GetServiceAsync(typeof(SVsTextManager)).ConfigureAwait(true);
```

Comment:
Small question:
Isn't `var` always assumes a nullable reference type? Why/when the cast can change the behavior?

Reply:
The problem here is entirely constrained to the right hand side. `object?` cannot be explicitly cast to `object` without a warning or a suppression.

Reply:
@sharwell Thanks a lot for clarifying